pkg/api/v20231122/openshiftcluster_validatestatic.go

```diff
@@ -35,11 +35,12 @@ func (sv openShiftClusterStaticValidator) Static(_oc interface{}, _current *api.
 	sv.domain = domain
 	sv.requireD2sV3Workers = requireD2sV3Workers
 	sv.resourceID = resourceID
-
+	var architectureVersion *api.ArchitectureVersion
```

Comment:
I am using a pointer for the new variable `architectureVersion`, I chose to do it this way as there is no Architecture version on cluster create (because there is no `_current` cluster).  Also, I didn't think it would be a good idea to use the default 0 value for the variable as it could be mistaken for ArchitectureVersionV1